### PR TITLE
Added vertical flipping in bbox_flip

### PR DESCRIPTION
**Pull request by dhananjaisharma10**
Added vertical bounding box flipping for the `bbox_flip` function in `mmdet/datasets/transforms.py`. I needed this for implementing shearX/Y functionality. Hope it helps!

### CODE REVIEW

**Comment by hellock** [comment]
Thanks for adding the feature! Could you fix the linting error?

**Comment by dhananjaisharma10** [comment]
Hi!
I fixed the linting errors. As an aside, I have a suggestion: the `bbox_flip` function should, in my opinion, be moved to `mmcv/mmcv/image/transforms/geometry.py` along with `bbox_clip` and `bbox_scaling`.
Hope it helps!

**Comment by hellock** [comment]
Yes I agree. Some bbox operations are better to be included in mmcv.

File: mmdet/datasets/transforms.py [hunk 1]
```diff
@@ -10,7 +10,6 @@
 
 class ImageTransform(object):
     """Preprocess an image.
-
```
Comment (hellock) [inline mmdet/datasets/transforms.py]: The blank line between the summary and detailed description is better to be kept.

Comment (dhananjaisharma10) [reply mmdet/datasets/transforms.py]: I agree, but the blank line is not present in any of the classes in the same file. Please let me know the changes that I should make. Thanks!

Comment (hellock) [reply mmdet/datasets/transforms.py]: For the docstrings that contains a summary and detailed description, usually we add a blank line between them. You can just recover the changes in such docstrings.

Comment (dhananjaisharma10) [reply mmdet/datasets/transforms.py]: Made those changes. I hope it's okay now. Please let me know. 

Also, does `mmdetection` accept packages written with PIL? I haven't seen any so far. So I wanted to confirm.

Comment (hellock) [reply mmdet/datasets/transforms.py]: Currently it works with numpy arrays. If you would like to play with PIL images, a wrapper may be needed.
```python
x = to_pil_img(x)
# some operations here
x = to_numpy_array(x)
```

Comment (hellock) [reply mmdet/datasets/transforms.py]: Seems that there are new linting errors.

Comment (dhananjaisharma10) [reply mmdet/datasets/transforms.py]: I recently wrote a package for Auto augmentation, but that makes use of PIL for certain transformations. Please let me know if I can contribute it with PIL or should I write it with cv2.

Comment (hellock) [reply mmdet/datasets/transforms.py]: I recommend using cv2, which is usually more efficient and has better compatibility with the current pipeline.

Comment (dhananjaisharma10) [reply mmdet/datasets/transforms.py]: Okay, I'll work on it and get back. Thanks for the response!

Comment (WangTianYuan) [reply mmdet/datasets/transforms.py]: > Okay, I'll work on it and get back. Thanks for the response!

@dhananjaisharma10 Hello, have you complete the Auto augmentation for mmdetection? I need to use it, but I don't know how to write it for mmdetection.

**Comment by openmmlab-v** [comment]
Hi @dhananjaisharma10 ！First of all, we want to express our gratitude for your significant PR in the mmdetection project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**